src/awkward/_typetracer.py

```diff
@@ -659,12 +659,10 @@ def copy(self):
         return self
 
     def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
-        self.touch_data()
-        replacements = [
-            numpy.empty(0, x.dtype) if hasattr(x, "dtype") else x for x in inputs
-        ]
-        result = getattr(ufunc, method)(*replacements, **kwargs)
-        return TypeTracerArray(result.dtype, shape=self._shape)
+        if method != "__call__" or len(inputs) == 0 or "out" in kwargs:
+            raise ak._errors.wrap_error(NotImplementedError)
+
+        return TypeTracer._module._generic_ufunc_func(ufunc)(*inputs, **kwargs)
```

Comment:
I am just looking at the diff here, in theory it's possible to have a `TypeTracerArray` without a `TypeTracer` instance being initialised. That would be wrong, but let's be defensive?

```suggestion
        return TypeTracer.instance()._module._generic_ufunc_func(ufunc)(*inputs, **kwargs)
```

Reply:
Good idea.